### ISSUE TITLE
Curl seem to be missing on Windows

### ISSUE THREAD

**Issue by Gbury** [ocaml/setup-ocaml#645]
Recent runs of setup-ocaml on windows have all failed with the following error:
```
 D:\cygwin\bin\bash.exe C:\hostedtoolcache\windows\opam\0.0.0.2\x64\opam64\install.sh --prefix /usr
  the following programs are not installed properly:
   curl
  I can't proceed :(
  Error: The process 'D:\cygwin\bin\bash.exe' failed with exit code 2
```
for a full log, see e.g. https://github.com/Gbury/dolmen/actions/runs/4195171274/jobs/7274335742#step:3:122

This may be related to #637 which occurred on unbuntu, while this error occurs on windows.

**Comment by emillon** [comment]
We're seeing the same on the dune CI.

**Comment by Gbury** [comment]
I'm not surprised, I guess that all CI workflows that use `setup-ocaml` on a windows runner will fail until this is resolved (or the image for the runner is changed to add `curl` ? I'm not sure what exactly caused the breakge we see).

**Comment by dra27** [comment]
The problem is that the runners have changed - the Cygwin cache is being extracted with the wrong `tar` command and symlinks relating to openssl are not being re-created correctly. What's surfacing as the error is that the `install.sh` tests that `curl` is working by attempting to download the headers for https://github.com and it's getting an ssl error.

From a debugging build on my fork which shows what `install.sh` is actually doing:
```
+ /usr/bin/timeout -s SIGTERM -k 1s 30.0s curl --insecure --retry 2 --head https://github.com/
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
  
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  curl: (27) SSL: couldn't create a context: error:0E079065:configuration file routines:def_load_bio:missing equal sign
  + add_failed curl
  + '[' -z '' ']'
  + failed=curl
  + exit_failed
  + '[' -n curl ']'
  + echo 'the following programs are not installed properly:'
  the following programs are not installed properly:
  + echo ' curl'
   curl
  + echo 'I can'\''t proceed :('
  I can't proceed :(
  + exit 2
```

For the different commands, cf. https://github.com/ocaml/dune/actions/runs/4129595063/jobs/7135368307#step:3:32 a couple of weeks ago:

```
Retrieve the Cygwin cache
  Received 121634816 of 492596606 (24.7%), 113.7 MBs/sec
  Received 3145700 of 492596606 (63.9%), 148.1 MBs/sec
  Received 492596606 of 492596606 (100.0%), 95.2 MBs/sec
  Cache Size: ~470 MB (492596606 B)
  C:\msys64\usr\bin\tar.exe --use-compress-program "zstd -d" -xf D:/a/_temp/19c2e7f6-5de2-436c-acc9-3c396a40cf1f/cache.tzst -P -C D:/a/dune/dune --force-local
```
Note use of `C:\msys64\usr\bin\tar.exe` versus now, cf. https://github.com/ocaml/dune/actions/runs/4226778525/jobs/7340585620#step:3:33:

```
Retrieve the Cygwin cache
  Received 100663296 of 527393390 (19.1%), 95.1 MBs/sec
  Received 8435456 of 5393390 (50.9%), 127.4 MBs/sec
  Received 444596224 of 527393390 (84.3%), 140.8 MBs/sec
  Received 527393390 of 527393390 (100.0%), 87.3 MBs/sec
  Cache Size: ~503 MB (527393390 B)
  C:\Windows\System32\tar.exe -z -xf D:/a/_temp/22d93640-5165-495f-9516-aa6b8dff/cache.tgz -P -C D:/a/dune/dune
```
Note use now of `C:\Windows\System32\tar.exe`.

I haven't investigated further as to whether it's an image change (with `PATH` having changed, or even MSYS2 no longer being installed there?) or a change in the actions/cache or whichever library does it. 

Note that in opam's workflows, we avoid this by going through a mildly convoluted process so that we have Cygwin's `tar.exe` with its required DLLs extracted earlier in the process, cf. https://github.com/ocaml/opam/blob/master/.github/scripts/cygwin.cmd#L138

**Comment by smorimoto** [comment]
This issue is fixed in the latest `v2.0.10` and `v2` tags.
